packages/shell-api/src/shell-internal-state.ts

```diff
@@ -0,0 +1,238 @@
+import {
+  AggregationCursor,
+  CursorIterationResult,
+  Cursor,
+  Help,
+  Database,
+  Mongo,
+  ReplicaSet,
+  Shard,
+  signatures,
+  ShellBson,
+  toIterator,
+} from './index';
+import { EventEmitter } from 'events';
+import { DatabaseOptions, Document, ServiceProvider, ReplPlatform } from '@mongosh/service-provider-core';
+import { MongoshInvalidInputError, MongoshUnimplementedError } from '@mongosh/errors';
+import AsyncWriter from '@mongosh/async-rewriter';
+
+/**
+ * Anything to do with the internal shell state is stored here.
+ */
+export default class ShellInternalState {
```

Comment:
I've moved the utilities that are static into a `helper.ts` file, but `emitApiCall` is a helper that uses the internal state (i.e. the current bus) so I'm not sure that should be removed from the internal class.

The top-level context methods like use/show/it are all state dependent. The reason they're in internal state is because we need to bind the method to whatever the currentDb/currentCursor/etc are at the moment. To me that feels like exactly what this class is for